### ISSUE TITLE
confusion about ResampleCloudSpatially

### ISSUE THREAD

**Issue by wangjue0706** [CloudCompare/CCCoreLib#101]
Hello, I would like to ask for some help with the problems I encountered while reading the code. I have two problems while trying to reproduce the **ResampleCloudSpatially** process. 
Firstly, compared to using the cloudcompare software, I spend too much time using **getPointsInSphericalNeighbourhood** on the same computer, especially on **getCellIndex**. For example, the software takes 200ms, while the source code takes 1000ms, I would like to ask what causes the difference between the two, could it be multithreading? 
Secondly, regarding the use of **ADAPTATIVE_BINARY_SEARCH** in **getCellIndex**, I know it's a binary search, but when running the code
```
float centralPoint = 0.5f + 0.75f*(static_cast&lt;float&gt;(truncatedCellCode-beginCode)/(-0.5f)); //0.75 = speed coef (empirical)
unsigned middle = begin + static_cast&lt;unsigned&gt;(centralPoint*float(end-begin));
```
centralPoint can be a large negative number, causing an exception in middle. Is there a problem with this code?"

**Comment by dgirardeau** [comment]
So first, I believe you may be compiling the code in Debug mode on your side (which is much slower than Release).

And then, yes this code works, as long as the cell indexes are unique and sorted in the ascending order. Mind that you can simply re-use CCCoreLib in a separate project (its license is LGPL also).

**Comment by wangjue0706** [comment]
Thank you for your answer, but I still have doubts about the second question. I tried to use getCellIndex(cellcode, bitshift, begin, end) instead of getCellIndex(cellcode, bitshift) for the search operation, but it crashed at this point when I used it. The picture is the information I got when debugging, and it seems that there is indeed a problem with the calculation of centralPoint and middle.
![Desktop Screenshot 2024 04 27 - 22 13 55 21](https://github.com/CloudCompare/CCCoreLib/assets/64588803/95a93c7c-0e90-4f2f-96b6-823e00beb2d9)
